packages/react/src/components/Storage/FileUploader/FileUploader.tsx

```diff
@@ -157,33 +133,35 @@ export function FileUploader({
       const statuses = [...fileStatuses];
       const status = fileStatuses[index];
 
-      statuses[index] = { ...status, paused: false };
+      statuses[index] = { ...status, fileState: null, isLoading: true };
       setFileStatuses(statuses);
     };
   };
+
   const onFileClick = () => {
```

Comment:
all these event handler functions should be memoized with `useCallback` hook for improved performance. Otherwise React will re-instantiate the eventHandler function on each render and child components will re-render every time FileUploader re-renders.

Reply:
OK will do! Updated in [d6516d7](https://github.com/aws-amplify/amplify-ui/pull/2892/commits/d6516d733d4a1a9a9cb696c3796625d309142a6c)